### PR TITLE
App name validation

### PR DESCRIPTION
**Pull request by siddhantsorann**
Related to [issue](https://github.com/spring-cloud/spring-cloud-dataflow/issues/4090)
Adding validation for application name while receiving API Request to register a new API.

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java [hunk 1]
```diff
@@ -225,6 +225,8 @@ public void register(@PathVariable("type") ApplicationType type, @PathVariable("
 			@RequestParam("uri") String uri, @RequestParam(name = "metadata-uri", required = false) String metadataUri,
 			@RequestParam(value = "force", defaultValue = "false") boolean force) {
 
+		if (!isAppNameValid(name))
+			throw new IllegalArgumentException("Application name: " + name + " cannot contain ':'");
```
Comment (ntantri) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java]: This `isAppNameValId` if gets used for other wildcard validations, then we may fall short on the error message - like cannot contain  `:`.
Maybe declare a wildcard constant and check with contains with the same constant.

**Review by ntantri — CHANGES_REQUESTED** [review]

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java (hunk 1 above)
Comment (siddhantsorann) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java:229]: That makes sense, currently, I'm checking exactly what all validation would the project owners want, will update this accordingly.

Comment (siddhantsorann) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java]: Updated the validation method.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java [hunk 2]
```diff
@@ -437,6 +438,16 @@ private void prefetchMetadata(List<AppRegistration> appRegistrations) {
...
 
+	private void validateApplicationName(String name) {
+		char[] invalidWildCards = new char[]{':'};
+		for (char invalidWildCard : invalidWildCards) {
+			if (name.contains(Character.toString(invalidWildCard))) {
+				throw new IllegalArgumentException(
+						"Application name: " + name + " cannot contain '" + invalidWildCard + "'");
+			}
+		}
+	}
```
Comment (ntantri) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java:453]: Maybe just do all validation in one shot?
```java
private void validateApplicationName(String name) {
  char[] invalidWildCards = new char[] {
    ':'
  };
  
  string invalidChars = "";
  for (char invalidWildCard: invalidWildCards) {
    if (name.contains(Character.toString(invalidWildCard))) {
      invalidChars += "'" + invalidWildCard + "' ";
    }
  }

  if (invalidChars != "") {
    throw new IllegalArgumentException("Application name: " + name + " cannot contain: " + invalidChars);
  }
}
```

**Review by ntantri — CHANGES_REQUESTED** [review]

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java (hunk 2 above)
Comment (siddhantsorann) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java:453]: This is a good point, updated.

**Review by ntantri — APPROVED** [review]

**Pull request merged**